tests/unit/restapi/lib/helpers.py

```diff
@@ -30,3 +31,86 @@ def is_iso_format(date_string: str) -> bool:
 
     except ValueError:
         return False
+
+
+def validate_non_ref_feilds(
+    expected_keys: set[str],
+    response: dict[str, Any], 
+    expected_contents: dict[str, Any],
+) -> bool:
+    try:
+        base_resource_keys: set[str] = {
+            "id",
+            "snapshotId",
+            "createdOn",
+            "lastModifiedOn",
+            "latestSnapshot",
+        }
+        for key in base_resource_keys:
+            assert key in expected_keys
+    except ValueError("Base resource keys are missing"):
+        return False
+    
+    try:
+        assert isinstance(response["id"], int)
+        assert isinstance(response["snapshotId"], int)
+        assert isinstance(response["createdOn"], str)
+        assert isinstance(response["lastModifiedOn"], str)
+        assert isinstance(response["latestSnapshot"], bool)
+    except ValueError("Response keys do not match expected keys types"):
+        return False
+    
+    try:
+        assert is_iso_format(response["createdOn"])
+        assert is_iso_format(response["lastModifiedOn"])
+    except ValueError("Required response keys are not in iso format"):
+        return False
+    
+    try:
+        if "name" in expected_keys:
+            assert isinstance(response["name"], str)
+            assert response["name"] == expected_contents["name"]
+        if "description" in expected_keys:
+            assert isinstance(response["description"], str)
+            assert response["description"] == expected_contents["description"]
+    except ValueError("Mutable fields keys do not match expected key types or values"):
+        return False
+    
+    return True
+
+
+def validate_group_ref(
+    group_ref: dict[str, Any], expected_contents: dict[str, Any]
+) -> bool:
+    try:
+        assert isinstance(group_ref["id"], int)
+        assert isinstance(group_ref["group"]["name"], str)
+        assert isinstance(group_ref["group"]["url"], str)
+        assert group_ref["id"] == expected_contents["group"]["id"]
+        return True
```

Comment:
`group_ref` doesn't have a `"group"` key. Should just be `group_ref["name"]` etc.